src/System.Text.Primitives/System/Text/FormattingData.cs

```diff
@@ -93,6 +385,75 @@ public static FormattingData InvariantUtf8
             }
         }
 
+        /// <summary>
+        /// Parse the next byte in a byte array. Will return either a DigitOrSymbol value, an InvalidCharacter, or a Continue
+        /// </summary>
+        /// <param name="nextByte">The next byte to be parsed</param>
+        /// <param name="bytesParsed">The total number of bytes parsed (will be zero until a code unit is deciphered)</param>
+        /// <returns></returns>
+        public byte ParseNextByte(byte nextByte, out int bytesParsed, ref int nodeIndex, ref int level)
```

Comment:
Why not have an overload (or just one method) that takes a buffer, instead of having state machine APIs exposed publicly?